internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -334,12 +335,26 @@ func resourceApiManagementApi() *pluginsdk.Resource {
 				Optional: true,
 			},
 		},
+
+		CustomizeDiff: pluginsdk.CustomDiffWithAll(
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				if d.Get("version").(string) != "" && d.Get("version_set_id").(string) == "" {
```

Comment:
Looks like you saw the errors around 
```
 * setting `version` without the required `version_set_id`
```
That's because `version_set_id` is interpolated with `azurerm_api_management_api_version_set.test.id` and Terraform doesn't know what an interpolated value is at plan time so it marks it as an empty string. 

What you can instead do is use `d.GetRawConfig().AsValueMap()["version_set_id"].IsNull()` which Terraform can look to see if it's interpolated or not.

You can make similar changes to the check below as well